src/features/events/store.ts

```diff
@@ -166,6 +166,26 @@ const eventsSlice = createSlice({
       item.data = event;
       item.isLoading = false;
       item.loaded = new Date().toISOString();
+
+      if (event.campaign) {
+        if (!state.eventsByCampaignId[event.campaign.id]) {
+          state.eventsByCampaignId[event.campaign.id] =
+            remoteList<ZetkinEvent>();
+          state.eventsByCampaignId[event.campaign.id].items.push(
+            remoteItem(event.id, { data: event })
+          );
+        }
+
+        const eventItem = state.eventsByCampaignId[
+          event.campaign.id
+        ].items.find((item) => item.id == event.id);
+        if (eventItem) {
+          eventItem.data = { ...eventItem.data, ...event };
+          eventItem.mutating = [];
+        }
+        state.eventsByCampaignId[event.campaign!.id].loaded =
+          new Date().toISOString();
```

Comment:
This is problematic because it's lying and saying that the full list for this campaign has loaded, when that's not the case. Just a single event has loaded, and if some part of the UI requests this list in the future, it needs to load in full. Currently, it would not loaded (because `shouldLoad()` thinks it's already loaded) and it would look in the UI like the campaign has just a single event.